### PR TITLE
testing/libspatialite: new aport

### PR DESCRIPTION
**Pull request by chambbj**
https://www.gaia-gis.it/fossil/libspatialite/index
SpatiaLite is an open source library intended to extend the SQLite core to
support fully fledged Spatial SQL capabilities.

### CODE REVIEW

**Comment by chambbj** [comment]
@ncopa This PR may have been overlooked as there was initially a typo and there was no package found during the Travis build. Everything should be good now.

File: testing/libspatialite/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+install=""
+subpackages="$pkgname-dev"
+source="$pkgname-$pkgver.tar.gz::http://www.gaia-gis.it/gaia-sins/$pkgname-$_pkgver.tar.gz"
+builddir="$srcdir/$pkgname-$_pkgver"
+
+build() {
+        cd "$builddir"
+
+        ./configure \
+                -prefix=/usr \
```
Comment (fabled) [inline testing/libspatialite/APKBUILD]: This should be `--prefix` ?
We also normally have `--build=$CBUILD` and `--host=$CHOST`.

File: testing/libspatialite/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+pkgname=libspatialite
+pkgver=4.3.0
+_pkgver="$pkgver"a
+pkgrel=0
+pkgdesc="SpatiaLite is an open source library intended to extend the SQLite core to support fully fledged Spatial SQL capabilities."
+url="https://www.gaia-gis.it/fossil/libspatialite/index"
+arch="all"
+license="MPLv1.1"
+options="!check"
+makedepends="automake geos-dev sqlite-dev proj4-dev zlib-dev libxml2-dev"
```
Comment (fabled) [inline testing/libspatialite/APKBUILD]: Is `automake` required? You don't seem to regenerate the autoconfig things, so this might be not needed.

Comment (chambbj) [reply testing/libspatialite/APKBUILD]: No. Maybe a copy/paste error. Will fix.

File: testing/libspatialite/APKBUILD (hunk 1 above)
Comment (chambbj) [reply testing/libspatialite/APKBUILD]: I'm sure I inadvertently removed the missing `-`. Surprised I didn't notice this earlier, though perhaps it just resulted in a warning that I overlooked.

As for `--build` and `--host`, setting these result in errors:
```
checking build system type... Invalid configuration `x86_64-alpine-linux-musl': machine `x86_64-alpine-linux' not recognized
checking host system type... Invalid configuration `x86_64-alpine-linux-musl': machine `x86_64-alpine-linux' not recognized
```

Things build fine without.

Comment (fabled) [reply testing/libspatialite/APKBUILD]: They are needed for potential cross build support. The above error is fixed by using `update_config_sub` in `prepare`.

**Comment by chambbj** [comment]
@fabled Changes made. Seems to build fine on my dev VM. Thanks again!

**Pull request closed without merging**